pkg/storage/vault/vault.go

```diff
@@ -0,0 +1,23 @@
+package vault
+
+import (
+	"github.com/scratchdata/scratchdata/config"
+	"github.com/scratchdata/scratchdata/pkg/storage/vault/aws"
+	"github.com/scratchdata/scratchdata/pkg/storage/vault/memory"
+)
+
+type Vault interface {
+	GetCredential(name string) (string, error)
+	SetCredential(name, value string) error
+}
+
+func NewVault(vaultConf config.Vault, destinations []config.Destination) (Vault, error) {
+	switch vaultConf.Type {
+	case "memory":
+		return memory.NewMemoryVault(destinations)
+	case "aws":
+		return aws.NewAWSVault(vaultConf.Settings, destinations)
```

Comment:
Do not pass `destinations` to this function.

Reply:
@poundifdef we need to pass it to the memory case, right ?